packages/next/build/swc/src/styled_jsx/mod.rs

```diff
@@ -0,0 +1,786 @@
+}
+
+fn get_existing_class_name(el: &JSXOpeningElement) -> (Option<Expr>, Option<usize>, Option<usize>) {
+  let mut spreads = vec![];
+  let mut class_name_expr = None;
+  let mut existing_index = None;
+  let mut existing_spread_index = None;
+  for i in (0..el.attrs.len()).rev() {
+    match &el.attrs[i] {
+      JSXAttrOrSpread::JSXAttr(JSXAttr {
+        name: JSXAttrName::Ident(Ident { sym, .. }),
+        value,
+        ..
+      }) => {
+        if sym == "className" {
+          existing_index = Some(i);
+          class_name_expr = match value {
+            Some(JSXAttrValue::Lit(str_lit)) => Some(Expr::Lit(str_lit.clone())),
+            Some(JSXAttrValue::JSXExprContainer(JSXExprContainer {
+              expr: JSXExpr::Expr(expr),
+              ..
+            })) => Some(*expr.clone()),
+            None => None,
+            _ => None,
+          };
+          break;
+        }
+      }
+      JSXAttrOrSpread::SpreadElement(SpreadElement { expr, .. }) => {
+        if let Expr::Object(ObjectLit { props, .. }) = &**expr {
+          let mut has_spread = false;
+          let mut has_class_name = false;
+          for j in 0..props.len() {
+            if let PropOrSpread::Prop(prop) = &props[j] {
+              if let Prop::KeyValue(KeyValueProp { key, value }) = &**prop {
+                if let PropName::Ident(Ident { sym, .. }) = key {
+                  if sym == "className" {
+                    has_class_name = true;
+                    class_name_expr = Some(*value.clone());
+                    if props.len() == 1 {
+                      existing_spread_index = Some(i);
+                    }
+                  }
+                }
+              }
+            } else {
+              has_spread = true;
+            }
+          }
+          if has_class_name {
+            break;
+          }
+          if !has_spread {
+            continue;
+          }
+        }
+
+        let valid_spread = match &**expr {
+          Expr::Member(_) => true,
+          Expr::Ident(_) => true,
+          _ => false,
+        };
+
+        if valid_spread {
+          let member_dot_name = Expr::Member(MemberExpr {
+            obj: ExprOrSuper::Expr(Box::new(*expr.clone())),
+            prop: Box::new(Expr::Ident(ident("className"))),
+            span: DUMMY_SP,
+            computed: false,
+          });
+          // `${name} && ${name}.className != null && ${name}.className`
+          spreads.push(and(
+            and(
+              *expr.clone(),
+              not_eq(
+                member_dot_name.clone(),
+                Expr::Lit(Lit::Null(Null { span: DUMMY_SP })),
+              ),
+            ),
+            member_dot_name.clone(),
+          ));
+        }
+      }
+      _ => {}
+    };
+  }
+
+  let spread_expr = match spreads.len() {
+    0 => None,
+    _ => Some(join_spreads(spreads)),
+  };
+
+  let class_name_expr = match class_name_expr {
+    Some(Expr::Tpl(_)) => Some(class_name_expr.unwrap()),
```

Comment:
This can be written as

```rust
Some(e @ Expr::Tpl(_)) => Some(e),
```

It's just a coding style issue, but `unwrap` makes debugging with the eye harder.